src/lib/tasks/server/server_task.hpp

```diff
@@ -1,16 +1,22 @@
 #pragma once
 
+#include <boost/thread/future.hpp>
+
 #include "scheduler/abstract_task.hpp"
-#include "server/hyrise_session.hpp"
 
 namespace opossum {
 
+template<typename T>
 class ServerTask : public AbstractTask {
  public:
-  explicit ServerTask(std::shared_ptr<HyriseSession> session) : _session(std::move(session)) {}
+  explicit ServerTask(){}
+  
+  using result_type = T;
```

Comment:
we don't use this, so it can be removed

Reply:
It's being used in HyriseSession::_dispatch_server_task